src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,577 @@
+        </tr>
+    </table>
+</CENTER>
+
+    @em COMET must be installed before this wrapper can be used. This wrapper
+    has been successfully tested with several versions of COMET.
+
+    This adapter supports relative database filenames, which (when not found in the current working directory) is looked up in
+    the directories specified by 'OpenMS.ini:id_db_dir' (see @subpage TOPP_advanced).
+
+    COMET settings not exposed by this adapter can be directly adjusted using an txt param file, which can be generated using comet -p.
+    By default, All (!) parameters available explicitly via this param file will take precedence over the wrapper parameters.
+
+    Parameter names have been changed to match names found in XTandem or MSGFPlus Search Engines, however some are Comet specific.
+    For a detailed description of all available parameters check the Comet documentation at http://comet-ms.sourceforge.net/parameters/parameters_201601/
+
+    <B>The command line parameters of this tool are:</B>
+    @verbinclude TOPP_COMETAdapter.cli
+    <B>INI file documentation of this tool:</B>
+    @htmlinclude TOPP_COMETAdapter.html
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+
+class TOPPCOMETAdapter :
+  public TOPPBase
+{
+public:
+  TOPPCOMETAdapter() :
+    TOPPBase("COMETAdapter", "Annotates MS/MS spectra using COMET.")
+  {
+  }
+
+protected:
+  void registerOptionsAndFlags_()
+  {
+
+    registerInputFile_("in", "<file>", "", "Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerOutputFile_("out", "<file>", "", "Output file");
+    setValidFormats_("out", ListUtils::create<String>("idXML"));
+    registerInputFile_("database", "<file>", "", "FASTA file or pro file. Non-existing relative file-names are looked up via'OpenMS.ini:id_db_dir'", true, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("database", ListUtils::create<String>("FASTA"));
+    registerInputFile_("comet_executable", "<executable>",
+      // choose the default value according to the platform where it will be executed
+      "comet.exe",                     
+      "Comet executable of the installation e.g. 'comet.exe'", true, false, ListUtils::create<String>("skipexists"));
+
+    //
+    // Optional parameters //
+    //
+
+    //Files
+    registerOutputFile_("pin_out", "<file>", "", "Output file - for percolator input", false);
+    setValidFormats_("pin_out", ListUtils::create<String>("pin"));
+    registerInputFile_("default_params_file", "<file>", "", "Default Comet params file. All parameters of this take precedence. A template file can be generated using comet.exe -p", false, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("default_params_file", ListUtils::create<String>("txt"));
+    registerIntOption_("threads", "<num>", 1, "number of threads", false, true);
```

Comment:
this flag is added by TOPPBase anyway... not sure what happens if you add it again...